doc/development/05_repo_manager.md

```diff
@@ -0,0 +1,202 @@
+
+```rust
+/// Builds a RepositoryManager. This will use a disk based KeyValueStore using the
+/// the data directory specified in the supplied `Config`.
+pub fn build(config: Arc<Config>, signer: Arc<KrillSigner>) -> Result<Self, Error> { ... }
+```
+
+When the `RepositoryManager` is created, it still needs to be initialized. This
+is done this way, because we cannot change the base URIs of an operational RPKI
+repository. There is no way to inform publishing CAs that the location of their
+objects has changed. Therefore we have an explicit function for this:
+
+```rust
+/// Create the publication server, will fail if it was already created.
+pub fn init(&self, uris: PublicationServerUris) -> KrillResult<()> {
+    info!("Initializing repository");
+    self.access.init(uris.clone(), &self.signer)?;
+    self.content.init(&self.config.data_dir, uris)?;
+    self.content.write_repository(&self.config.repository_retention)?;
+
+    Ok(())
+}
+```
+
+Note that it writes the empty repository as part of the initialization process.
+We do this, because that gives users an opportunity to verify that the repository
+content can be accessed. In particular: expect an RRDP notification.xml and a snapshot.xml
+containing 0 entries to be published.
+
+If it is found that the repository is not set up correctly, then it can be 'cleared'
+so it can be initialized again:
+
+```rust
+/// Clear the publication server. Will fail if it still
+/// has publishers. Or if it does not exist
+pub fn repository_clear(&self) -> KrillResult<()> {
+    self.access.clear()?;
+    self.content.clear()
+}
+```
+
+
+Adding / Removing Publishers
+----------------------------
+
+To add a publisher we have to submit an `rfc8183::PublisherRequest`, and an `Actor`
+so that we attribute the change in the history of the `RepositoryAccess` component.
+The request type can be created from XML supplied by a remote publisher. The 'handle'
+is the local name that we will use for the publisher, and the XML contains a suggestion
+set by the remote party. But, it is possible (and perhaps recommended) to override
+this 'handle' in the request object with something that is unique to *us*, e.g. a UUID.
+
+```rust
+/// Adds a publisher. This will fail if a publisher already exists for the handle in the request.
+pub fn create_publisher(&self, req: rfc8183::PublisherRequest, actor: &Actor) -> KrillResult<()> {
+    let name = req.publisher_handle().clone();
+
+    self.access.add_publisher(req, actor)?;
+    self.content.add_publisher(name)
+}
+```
+
+The publisher is going to need to get RFC8183 Repository Response XML. We can get
+an `rfc8183::RepositoryResponse` which can be printed (`fmt::Display`) as XML:
+
+```rust
+/// Returns the RFC8183 Repository Response for the publisher
+pub fn repository_response(
+    &self,
+    rfc8181_uri: uri::Https,
+    publisher: &PublisherHandle,
+) -> KrillResult<rfc8183::RepositoryResponse> {
+    self.access.repository_response(rfc8181_uri, publisher)
```

Comment:
Is it correct that the constructed return value refers to RFC 8181 while the return type and function level comment both  refer to RFC 8183?